code/game/objects/effects/effect_system/fluid_spread/effects_foam.dm

```diff
@@ -416,6 +416,7 @@
 	alpha = 120
 	max_integrity = 10
 	pass_flags_self = PASSGLASS
+	var/list/ignored_gases = list(/datum/gas/nitrogen, /datum/gas/oxygen, /datum/gas/pluoxium, /datum/gas/halon)
```

Comment:
```suggestion
	var/static/list/ignored_gases = typecacheof(list(
		/datum/gas/nitrogen,
		/datum/gas/oxygen,
		/datum/gas/pluoxium,
		/datum/gas/halon,
	))
```
Until subtypes that override their parent definition don't exist, it may as well be static.